lib/client/kubesession.go

```diff
@@ -118,6 +121,28 @@ func NewKubeSession(ctx context.Context, tc *TeleportClient, meta types.SessionT
 	return s, nil
 }
 
+func kubeSessionNetDialer(ctx context.Context, tc *TeleportClient, kubeAddr string) client.ContextDialer {
+	dialOpts := []client.DialOption{
+		client.WithInsecureSkipVerify(tc.InsecureSkipVerify),
+	}
+
+	// Add options for ALPN connection upgrade only if kube is served at Proxy
+	// web address.
+	if tc.WebProxyAddr == kubeAddr && tc.TLSRoutingConnUpgradeRequired {
```

Comment:
`tc.WebProxyAddr == kubeAddr` 

I wonder about cases where this condition will be evaluated to false
AFIK kube sessions are using websocket conn so it must be a proxy addr.

Reply:
The websocket is served by the kube server instead of the web server
https://github.com/gravitational/teleport/blob/19b193acc012db095fb12bb82886370993fc3289/lib/kube/proxy/forwarder.go#L332
where TLS config uses the kube SNI prefix.

So the `false` case is when proxy in `separate` port mode and kube has its own public address.